datafusion/physical-optimizer/src/join_selection.rs

```diff
@@ -903,7 +900,7 @@ mod tests_statistical {
             let original_schema = join.schema();
 
             let optimized_join = JoinSelection::new()
-                .optimize(join.clone(), &ConfigOptions::new())
+                .optimize(Arc::<HashJoinExec>::clone(&join), &ConfigOptions::new())
```

Comment:
I think this can be simplified to the following (no need to explicitly spell out the type):


```suggestion
                .optimize(Arc::clone(&join), &ConfigOptions::new())
```

Similarly for the other changes in this file